src/cargo/core/compiler/unit_dependencies.rs

```diff
@@ -349,10 +349,20 @@ fn compute_deps<'a, 'cfg>(
                 .filter(|t| {
                     let no_required_features = Vec::new();
 
+                    let mut dep_features = HashSet::new();
```

Comment:
Can this code be shared with [`resolve_all_features`](https://github.com/rust-lang/cargo/blob/4c7a6389d5b1f3953d79a2c379691d81dbe415ef/src/cargo/ops/cargo_compile.rs#L944-L977).  There are some subtle issues, and I'd prefer not to duplicate it.

Reply:
What I can think of is to make `resolve_all_features` visible and call it from here instead of the testing unit features and dep features. Am I correct? Is there anything I am missing? I feel that I haven't made out what you mean.

Reply:
Yes, that seems correct.